compiler/rustc_expand/src/mbe.rs

```diff
@@ -17,23 +17,52 @@ use rustc_data_structures::sync::Lrc;
 use rustc_span::symbol::Ident;
 use rustc_span::Span;
 
-/// Contains the sub-token-trees of a "delimited" token tree, such as the contents of `(`. Note
-/// that the delimiter itself might be `NoDelim`.
+/// Contains the sub-token-trees of a "delimited" token tree such as `($var)`. The delimiter itself
+/// might be `NoDelim`.
 #[derive(Clone, PartialEq, Encodable, Decodable, Debug)]
 struct Delimited {
     delim: token::DelimToken,
-    tts: Vec<TokenTree>,
+    /// Note: This contains the opening and closing delimiters tokens (e.g. `(` and `)`), *unless*
+    /// `delim` is `NoDelim`. These token kinds must match `delim`, and the methods below
```

Comment:
`NoDelim` is not a *lack* of a delimiter, it's an *invisible* delimiter (https://doc.rust-lang.org/stable/proc_macro/enum.Delimiter.html#variant.None), it must never be dropped.
`rustc` parser dropping the `None` delimiter in some places is a bug (https://github.com/rust-lang/rust/issues/67062).

Could you at least keep it in `all_tts` and drop them as late as possible?
And add a FIXME linking to https://github.com/rust-lang/rust/issues/67062 in the place where they are dropped.